### PR TITLE
Add apiserver_total_request metric to the Prometheus longterm instance

### PR DESCRIPTION
**Pull request by jguipi**
**How to categorize this PR?**

/area monitoring
/kind enhancement

**What this PR does / why we need it**:

This PR Federate apiserver_total_request metric to the Prometheus longterm instance

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
Federate apiserver_total_request metric to the Prometheus longterm instance
```

### CODE REVIEW

File: pkg/component/observability/monitoring/prometheus/longterm/scrapeconfigs.go [hunk 1]
```diff
@@ -61,6 +61,7 @@ func CentralScrapeConfigs() []*monitoringv1alpha1.ScrapeConfig {
 						`{__name__="garden_seed_usage"}`,
 						`{__name__="garden_seed_capacity"}`,
 						`{__name__="etcdbr_snapshot_duration_seconds_count"}`,
+						`{__name__="apiserver_request_total", job=~="(virtual-garden-kube-apiserver)"}`
```
Comment (oliver-goetz) [inline pkg/component/observability/monitoring/prometheus/longterm/scrapeconfigs.go]:
```suggestion
						`{__name__="apiserver_request_total", job=~="(virtual-garden-kube-apiserver)"}`,
```

File: pkg/component/observability/monitoring/prometheus/longterm/scrapeconfigs_test.go [hunk 2]
```diff
@@ -62,6 +62,7 @@ var _ = Describe("PrometheusRules", func() {
 								`{__name__="garden_seed_usage"}`,
 								`{__name__="garden_seed_capacity"}`,
 								`{__name__="etcdbr_snapshot_duration_seconds_count"}`,
+								`{__name__="apiserver_request_total", job=~="(virtual-garden-kube-apiserver)"}`
```
Comment (oliver-goetz) [inline pkg/component/observability/monitoring/prometheus/longterm/scrapeconfigs_test.go]:
```suggestion
								`{__name__="apiserver_request_total", job=~="(virtual-garden-kube-apiserver)"}`,
```

**Review by oliver-goetz — CHANGES_REQUESTED** [review]
😁

File: pkg/component/observability/monitoring/prometheus/longterm/scrapeconfigs.go (hunk 1 above)
Comment (timuthy) [reply pkg/component/observability/monitoring/prometheus/longterm/scrapeconfigs.go]: Shouldn't it be?
```suggestion
						`{__name__="apiserver_request_total", job=~"(virtual-garden-kube-apiserver)"}`,
```

In general I'd vote for:
```suggestion
						`{__name__="apiserver_request_total", job="virtual-garden-kube-apiserver"}`,
```

Comment (oliver-goetz) [reply pkg/component/observability/monitoring/prometheus/longterm/scrapeconfigs.go]: oh yes 😅

File: pkg/component/observability/monitoring/prometheus/longterm/scrapeconfigs_test.go [hunk 3]
```diff
@@ -62,6 +62,7 @@ var _ = Describe("PrometheusRules", func() {
 								`{__name__="garden_seed_usage"}`,
 								`{__name__="garden_seed_capacity"}`,
 								`{__name__="etcdbr_snapshot_duration_seconds_count"}`,
+								`{__name__="apiserver_request_total", job=~="(virtual-garden-kube-apiserver)"}`,
```
Comment (timuthy) [inline pkg/component/observability/monitoring/prometheus/longterm/scrapeconfigs_test.go]:
```suggestion
								`{__name__="apiserver_request_total", job="virtual-garden-kube-apiserver"}`,
```

**Pull request merged**